### PR TITLE
feat(werklijst): add clickable rows with navigation and a11y support

### LINKED ISSUE #503: 🛠 Task: Klikbare rijen met navigatie
# 🛠 Task: Klikbare rijen met navigatie

---

## Intent

Als Coördinator of Behandelaar wil ik een contactverzoek openen door op de rij te klikken, zodat ik snel kan navigeren zonder een aparte knop te zoeken.

---

## Functional Description

De hele rij van een contactverzoek in de werklijst (zowel "Alle contactverzoeken" als "Mijn werkvoorraad") wordt klikbaar en opent het contactverzoekdetailscherm. De huidige "Klik hier"-link wordt vervangen door een "→"-teken. De rij krijgt een visueel hover-effect (achtergrondkleurwijziging + pointer-cursor) zodat duidelijk is dat de rij interactief is.

**Toegankelijkheid (a11y):** De implementatie volgt WCAG 2.1 richtlijnen. De "→" in de cel blijft een echte `<router-link>` met een beschrijvend `aria-label` (bijv. "Open contactverzoek [klantnaam]") zodat schermlezers het doel kunnen aankondigen. De rijklik wordt afgehandeld via een click-handler op de `<tr>` die programmatisch navigeert; de link in de rij blijft focusbaar en activeerbaar via toetsenbord (Enter/Space). Er worden geen extra ARIA-rollen op de `<tr>` gezet die de tabel-semantiek breken.

---

## Design Reference

N/A — geen aparte Figma. Volgt NLDS/Utrecht: hover-feedback via Utrecht design tokens. Pijl "→" als visuele indicator conform Feature #402 specificatie.

---

### In scope

- Klikbare rij op beide werklijsttabellen (`AllInterneTakenTable.vue`, `MyInterneTakenTable.vue`)
- Vervanging "Klik hier" door "→" met beschrijvend `aria-label`
- Hover-effect op rij (achtergrondkleur + `cursor: pointer`)
- Programmatische navigatie via `useRouter().push()` bij klik op de rij
- Toetsenbordtoegankelijkheid: link blijft focusbaar, Enter/Space activeert navigatie
- Aanpassing E2E test-locators (`Locators.cs`) voor nieuwe "→"-tekst

### Out of scope

- Sorteren op kolomhoofden (→ apart task)
- Filteren op "alleen verlopen" (→ apart task)
- Wijzigingen aan de detailpagina zelf

---

### Request

Geen wijziging — puur frontend.

### Response

Geen wijziging — puur frontend.

### Events / Side Effects

Geen — navigatie-interactie zonder state-wijziging.

---

## Technical Approach

- **Architecture:** Rij-klik via `@click`-handler op `<utrecht-table-row>` die `router.push()` aanroept. De bestaande `<router-link>` in de "Details"-cel wordt behouden als toegankelijke link met `aria-label`, tekst verandert van "Klik hier" naar "→".
- **Data flow:** Klik op rij → `router.push('/contactmoment/{nummer}')` → bestaande `ContactverzoekDetailView.vue`
- **A11y-aanpak:** Geen `role`-wijzigingen op `<tr>` (tabel-semantiek behouden). Link in cel fungeert als primair toetsenbord-navigatie-element. `aria-label` op link met contextinfo (bijv. klantnaam/onderwerp). Click-handler op rij voorkomt dubbele navigatie door `event.target` te checken — als de klik op de link zelf plaatsvindt, laat de `<router-link>` het afhandelen.
- **Styling:** SCSS scoped styles op de tabelcomponenten. Hover-effect via `:hover` pseudo-class met Utrecht design tokens. `cursor: pointer` op interactieve rijen (niet op lege-staat rij).
- **E2E impact:** `Locators.cs` `GetDetailsLink` methode moet aangepast worden van `HasText = "Klik hier"` naar `HasText = "→"`. Andere E2E tests die via de detaillink navigeren moeten ook aangepast worden.

### Aggregates & Invariants

- Aggregates: Geen — puur UI-interactie
- Invariants: Navigatie-doelpad is consistent met het huidige routerpatroon (`/contactmoment/{nummer}`)

---

## Impacted Areas

- Backend: Geen
- Frontend: `AllInterneTakenTable.vue`, `MyInterneTakenTable.vue`
- Database: Geen
- APIs: Geen
- External systems: Geen
- Tests: `Locators.cs`, `ContactverzoekScenarios.cs`

---

## Gherkin

```gherkin
Feature: Klikbare rijen met navigatie op werklijst

  Scenario: Klikken op een rij opent het contactverzoek
    Given een Behandelaar bekijkt "Mijn werkvoorraad" met minstens één contactverzoek
    When de Behandelaar op de rij van het contactverzoek klikt
    Then wordt het detailscherm van dat contactverzoek geopend

  Scenario: Klikken op een rij in "Alle contactverzoeken" opent het contactverzoek
    Given een Coördinator bekijkt "Alle contactverzoeken" met minstens één contactverzoek
    When de Coördinator op de rij van het contactverzoek klikt
    Then wordt het detailscherm van dat contactverzoek geopend

  Scenario: Details-kolom toont pijl in plaats van "Klik hier"
    Given een Behandelaar bekijkt de werklijst
    When de werklijst contactverzoeken toont
    Then bevat de Details-kolom het teken "→" in plaats van "Klik hier"

  Scenario: Rij toont visuele hover-feedback
    Given een Behandelaar bekijkt de werklijst met contactverzoeken
    When de Behandelaar met de muis over een rij beweegt
    Then verandert de achtergrondkleur van de rij
    And verandert de cursor naar een pointer

  Scenario: Rij-navigatie is toegankelijk via toetsenbord
    Given een Behandelaar bekijkt de werklijst met contactverzoeken
    When de Behandelaar via Tab naar de navigatielink in een rij navigeert
    And de Behandelaar op Enter drukt
    Then wordt het detailscherm van dat contactverzoek geopend

  Scenario: Navigatielink heeft een beschrijvend aria-label
    Given een Behandelaar bekijkt de werklijst met contactverzoeken
    When een schermlezer de navigatielink in een rij voorleest
    Then bevat het aria-label een beschrijving van het contactverzoek
```

---

## Edge Cases & Risks

- Klik op interactieve elementen binnen de rij (bijv. de link zelf) mag geen dubbele navigatie triggeren
- Lege werklijst-rij ("Geen contactverzoeken gevonden") mag niet klikbaar zijn
- Tekstselectie in een cel mag geen navigatie triggeren

### Edge Case Results

| Edge Case | Result | Bug Filed |
| --------- | ------ | --------- |
| Klik op interactieve elementen binnen de rij triggert geen dubbele navigatie | ✅ | — |
| Lege werklijst-rij is niet klikbaar | ✅ | — |
| Tekstselectie in een cel triggert geen navigatie | ✅ | — |

---

## Observability

- Logs: None required for this task
- Metrics: None required for this task
- Alerts: None required for this task

---

## Dependencies

Geen — onafhankelijk van sorteren en filteren.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: N/A — UI-only change
- Data migration: N/A

---

## Definition of Done

- [x] Linting passes (`npm run lint:ci`) — confirmed via PR #508 CI ("Lint Code Base" check: pass)
- [ ] Formatting passes (`npm run format:ci`)
- [x] All Gherkin scenarios correctly implemented (code inspection)
- [x] Edge cases handled
- [ ] PR approved by at least one reviewer
- [ ] Design reviewed (if UI)
- [ ] Documentation updated

---

## Test Mapping

Phase 1 (code inspection, this QA pass) — E2E automation is Phase 2, tracked in #509.

| Scenario | Result | Bug Filed |
| -------- | ------ | --------- |
| Klikken op een rij opent het contactverzoek | ✅ | — |
| Klikken op een rij in "Alle contactverzoeken" opent het contactverzoek | ✅ | — |
| Details-kolom toont pijl in plaats van "Klik hier" | ✅ | — |
| Rij toont visuele hover-feedback | ✅ | — |
| Rij-navigatie is toegankelijk via toetsenbord | ✅ | — |
| Navigatielink heeft een beschrijvend aria-label | ✅ | — |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

Make the entire row of a contactverzoek clickable to navigate to its detail page, replacing "Klik hier" with a centered "→" arrow. Adds hover feedback and follows WCAG 2.1 a11y guidelines.

### Changes

- **`use-row-navigation.ts`** — New composable with drag-detection guard (prevents navigation during text selection)
- **`main.scss`** — Global utility classes for clickable rows, centered details cell, and no-underline link
- **`AllInterneTakenTable.vue`** / **`MyInterneTakenTable.vue`** — Clickable rows with `@mousedown` + `@click`, "→" link with descriptive `aria-label`
- **`Locators.cs`** / **`ContactverzoekScenarios.cs`** — E2E test locator updates

### A11y

- `<router-link>` preserved as keyboard-focusable link with `aria-label`
- No ARIA roles added to `<tr>` — table semantics preserved
- Hover effect via CSS only (pointer + background)

Closes #503

### CODE REVIEW

File: InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/MyInterneTakenTable.vue [hunk 1]
```diff
@@ -36,9 +46,13 @@
...
-          <router-link :to="`/contactmoment/${taak?.aanleidinggevendKlantcontact?.nummer}`"
-            >Klik hier</router-link
+        <utrecht-table-cell class="details-cell">
+          <router-link
+            :to="`/contactmoment/${taak?.aanleidinggevendKlantcontact?.nummer}`"
+            :aria-label="`Open contactverzoek ${taak.aanleidinggevendKlantcontact?._expand?.hadBetrokkenen?.map((x) => x.volledigeNaam).find(Boolean) || taak.aanleidinggevendKlantcontact?.onderwerp || taak.aanleidinggevendKlantcontact?.nummer || ''}`"
+            class="details-link"
+            @click.stop
+            >→</router-link
           >
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/MyInterneTakenTable.vue:58]: Fixed in 282943c — link now uses `v-if="taak.aanleidinggevendKlantcontact?.nummer"`, row class is conditional, and click handler is guarded. Shows `-` placeholder when nummer is missing.

File: InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/AllInterneTakenTable.vue [hunk 2]
```diff
@@ -21,7 +23,13 @@
         <utrecht-table-cell colspan="8">Geen contactverzoeken gevonden</utrecht-table-cell>
       </utrecht-table-row>
 
-      <utrecht-table-row v-for="taak in interneTaken" :key="taak.uuid">
+      <utrecht-table-row
+        v-for="taak in interneTaken"
+        :key="taak.uuid"
+        class="clickable-row"
+        @mousedown="onRowMouseDown"
+        @click="navigateOnRowClick($event, `/contactmoment/${taak.contactmomentNummer}`)"
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/AllInterneTakenTable.vue]: Fixed in 282943c — `clickable-row` class is now conditional on `taak.contactmomentNummer` and the click handler is guarded.

File: InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/AllInterneTakenTable.vue [hunk 3]
```diff
@@ -46,21 +54,30 @@
...
-        <utrecht-table-cell>
-          <router-link :to="`/contactmoment/${taak.contactmomentNummer}`">Klik hier</router-link>
+        <utrecht-table-cell class="details-cell">
+          <router-link
+            :to="`/contactmoment/${taak.contactmomentNummer}`"
+            :aria-label="`Open contactverzoek ${taak.klantNaam || taak.onderwerp || taak.contactmomentNummer || ''}`"
+            class="details-link"
+            @click.stop
+            >→</router-link
+          >
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Client/src/components/interne-taken-tables/AllInterneTakenTable.vue:68]: Fixed in 282943c — link wrapped with `v-if="taak.contactmomentNummer"`, renders `-` placeholder otherwise.

File: InterneTaakAfhandeling.Web.Client/src/composables/use-row-navigation.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  const navigateOnRowClick = (e: MouseEvent, path: string) => {
+    // Let clicks on links behave as links, not row navigation.
+    if ((e.target as HTMLElement).closest("a")) return;
+
+    // Skip nav if the user was drag-selecting text (pointer moved >5px).
+    if (Math.abs(e.clientX - x) > 5 || Math.abs(e.clientY - y) > 5) return;
+
+    router.push(path);
+  };
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Client/src/composables/use-row-navigation.ts:27]: Fixed in 282943c — `start` is now `{ x, y } | null`, initialized as `null`. Drag check only applies when `start` is set, and resets after use.
